### PR TITLE
Supplementing pow tests

### PR DESCRIPTION
**Pull request by Madhav2310**
Supplementing pow tests files for #731

### CODE REVIEW

**Comment by faze-geek** [comment]
Maybe I am missing something, but why is `test_pow2.py` added in `tests/errors` ?  Is some kind of error message expected in` print(pow(2,2))` or `print(2**2)` ? Same for `test_pow3.py`

**Comment by certik** [comment]
LPython provides style suggestions for these, but those are not errors, so I think these should be just regular tests.

File: tests/errors/test_pow1.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+def f():
+    x: str
+    x = "a"
+    print(pow(x,3))
```
Comment (czgdp1807) [inline tests/errors/test_pow1.py:4]:
```suggestion
    print(pow(x,3))

```

File: tests/errors/test_pow3.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from ltypes import c32
+def f():
+    x: c32
+    x = 4 + 6j
+    print(pow(x,3))
```
Comment (czgdp1807) [inline tests/errors/test_pow3.py:5]:
```suggestion
    print(pow(x,3))

```

File: tests/errors/test_pow4.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+def f():
+    x: str
+    x = "t"
+    t: i64
+    t = pow(4,x)
+    print(t)
```
Comment (czgdp1807) [inline tests/errors/test_pow4.py:7]:
```suggestion
    print(t)

```

File: tests/errors/test_pow3.py (hunk 2 above)
Comment (czgdp1807) [reply tests/errors/test_pow3.py:5]: This should work. Its totally fine to raise complex numbers to some power. See a CPython example  below,

```python
Python 3.8.9 (default, Apr 13 2022, 08:48:06) 
[Clang 13.1.6 (clang-1316.0.21.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> (4 + 6j)**2
(-20+48j)
```

**Review by certik — APPROVED** [review]
This is fine with me. @czgdp1807 if this is ok with you, you can merge.

File: tests/reference/asr-test_pow4-ef60978.stderr [hunk 4]
```diff
@@ -0,0 +1,11 @@
+style suggestion: Could have used '**' instead of 'pow'
+ --> tests/errors/test_pow4.py:8:11
+  |
+8 |     print(pow(x, a, b))
+  |           ^^^^^^^^^^^^ '**' could be used instead
+
+semantic error: Arguments do not match for any generic procedure, pow
+ --> tests/errors/test_pow4.py:8:11
+  |
+8 |     print(pow(x, a, b))
```
Comment (czgdp1807) [inline tests/reference/asr-test_pow4-ef60978.stderr:10]: A better error message would be the following,

```
SemanticError: complex modulo
```

**Review by czgdp1807 — APPROVED** [review]

**Comment by czgdp1807** [comment]
I did a squash merge on this.

**Pull request merged**